Source/Core/Core/IOS/MIOS.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <cstring>
+#include <vector>
+
+#include "Common/CommonFuncs.h"
+#include "Common/Logging/Log.h"
+#include "Core/Boot/ElfReader.h"
+#include "Core/ConfigManager.h"
+#include "Core/DSPEmulator.h"
+#include "Core/HW/DSP.h"
+#include "Core/HW/Memmap.h"
+#include "Core/HW/SystemTimers.h"
+#include "Core/IOS/MIOS.h"
+#include "Core/PowerPC/PowerPC.h"
+#include "DiscIO/NANDContentLoader.h"
+
+namespace IOS
+{
+namespace HLE
+{
+namespace MIOS
+{
+constexpr u64 MIOS_TITLE_ID = 0x0000000100000101;
+
+// Source: https://wiibrew.org/wiki/ARM_Binaries
+struct ARMBinary final
+{
+  ARMBinary(const std::vector<u8>& bytes);
+
+  bool IsValid() const;
+  std::vector<u8> GetElf() const;
+  u32 GetHeaderSize() const;
+  u32 GetElfOffset() const;
+  u32 GetElfSize() const;
+
+private:
+  std::vector<u8> m_bytes;
+};
+
+ARMBinary::ARMBinary(const std::vector<u8>& bytes)
```

Comment:
You can use an initializer list here:

```cpp
ARMBinary::ARMBinary(const std::vector<u8>& bytes) : m_bytes(bytes)
{
}
```

You could also provide a move constructor alongside that constructor that would be like so:

```cpp
ARMBinary::ARMBinary(std::vector<u8>&& bytes) : m_bytes(std::move(bytes))
{
}
```

which would prevent needing to perform a full copy of the data in certain scenarios, e.g. `ARMBinary mios{GetMIOSBinary()};`